### PR TITLE
[POT]: Simplified PTQ API (prototype)

### PR DESCRIPTION
**Pull request by AlexKoff88**
### Details:
 - Added a prototype of simplified quantization API and an example that demonstrates it
 - The list of parameters can be extended if needed
 - API can be extended with the Metric for the Accuracy-aware scenario

### CODE REVIEW

**Comment by AlexKoff88** [comment]
cc'ed @alexsu52

File: tools/pot/openvino/tools/pot/api/samples/classification/simple_api_example.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    data_loader = ImageNetDataLoader(dataset_config)
+
+    # Define mandatory quantization parameters
+    parameters = QuantizationParameters()
+    parameters.model_name = 'sample_model'
+    parameters.model_path = os.path.expanduser(args.model),
+    parameters.weights_path = os.path.expanduser(args.weights)
+
+    # Quantize model
+    compressed_model = quantize_post_training(parameters, data_loader)
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/samples/classification/simple_api_example.py]: A good suggestion for the case when the model object is an input to the quantization code. But this is not the case for POT which cannot consume ov.Function and deals with a different representation. So, we need to pass a model path to the API which consists of the path to XML and the path to BIN. Of course, we can do the serialization of ov.Function under the hood and load the needed representation from the saved IR but it can take quite a lot of time and there can be some security issues with that. Moving POT to ov.Function and nGraph transformations should address this problem.

[the remark this answers is not in the document]
Comment (alexsu52) [reply tools/pot/openvino/tools/pot/api/samples/classification/simple_api_example.py]: Ok, It makes sense.

**Comment by AlexKoff88** [comment]
@helena-intel, can you please take a look as well?

File: tools/pot/openvino/tools/pot/api/helpers.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        """
+        Initializes quantization parameters with some reasonable verified defaults.
+        """
+        self._model_name = "model"
+        self._target_device = "ANY"
+        self._method_name = "DefaultQuantization"
+        self._preset = "performance"
+        self._subset_size = 300
+        self._model_type = None
+        self._fast_bias_correction = True
```
Comment (helena-intel) [inline tools/pot/openvino/tools/pot/api/helpers.py:24]: It would be nice if these properties can be set in the `__init__` method, so you can set them in one line, e.g. `parameters = QuantizationParameters(model_name="awesome_model",  preset="mixed")`. I do like having these sane defaults with nothing required.

**Review by helena-intel — COMMENTED** [review]
Thanks @AlexKoff88  this is nice. I already find POT easy to use, but this reduces some boilerplate. 

I saw the new samples are not linked yet in the README. Do you want to keep this low profile for now, and is the idea that this will be the default API in the future?

File: tools/pot/openvino/tools/pot/api/helpers.py [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+    compressed_model = pipeline.run(model)
+
+    return compressed_model
+
+class ExportParameters(object):
+    """
+    Optimized model exports specific parameters.
+    """
+    def __init__(self):
+        self._compress_weights = True
```
[the remark this answers is not in the document]
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/helpers.py:232]: GNA HW does not support compressed weights yet.

**Comment by AlexKoff88** [comment]
@helena-intel, thanks for the comments. This is a POC for now. BTW, I have some concerns regarding accuracy-aware quantization and I am going to revise its API soon. The main idea is to make it as smooth as possible for the user to apply this from the framework-based code.

File: tools/pot/openvino/tools/pot/api/helpers.py (hunk 2 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/helpers.py:24]: Good suggestion!

**Comment by MaximProshin** [comment]
@adrianboguszewski , as you had a comparison of TFLite quantization and POT with some recommendations from your side, I believe you'll be interested in proposed changes from Alex. Please take a look.

File: tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
+# Copyright (C) 2020-2022 Intel Corporation
```
Comment (MaximProshin) [inline tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py:1]: I think samples without references in POT documentation will not work. Do you plan to extend the documentation?

File: tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+# SPDX-License-Identifier: Apache-2.0
```
Comment (MaximProshin) [inline tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py:2]: Can we have some intro description at the beginning of each sample explaining its purpose, which algo is used, etc?

File: tools/pot/openvino/tools/pot/api/samples/object_detection/object_detection_simple_api.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        'images_path': os.path.expanduser(args.dataset),
+        'annotation_path': os.path.expanduser(args.annotation_path),
+    }
+
+    # Step 1: Initialize the data loader.
+    data_loader = COCOLoader(dataset_config)
+    # Step 2: Initialize the metric.
+    metric = MAP(91, data_loader.labels)
+
+    parameters = AccurracyAwareQuantizationParameters()
```
Comment (MaximProshin) [inline tools/pot/openvino/tools/pot/api/samples/object_detection/object_detection_simple_api.py:37]: Maybe it's worth to mention it as a separate step.

File: tools/pot/openvino/tools/pot/api/samples/object_detection/object_detection_simple_api.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+import os
+
+from openvino.tools.pot import AccurracyAwareQuantizationParameters, quantize_with_accuracy_control, \
+                                export
+from openvino.tools.pot.utils.logger import init_logger
+from openvino.tools.pot.api.samples.utils.argument_parser import get_common_argparser
+from openvino.tools.pot.api.samples.object_detection.metric import MAP
+from openvino.tools.pot.api.samples.object_detection.data_loader import COCOLoader
+
```
Comment (MaximProshin) [inline tools/pot/openvino/tools/pot/api/samples/object_detection/object_detection_simple_api.py:12]: 5 lines of imports, can it be also simplified somehow?

File: tools/pot/openvino/tools/pot/api/helpers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (adrianboguszewski) [reply tools/pot/openvino/tools/pot/api/helpers.py:232]: Agree, compres_weight should be parameter of constructor

File: tools/pot/openvino/tools/pot/api/helpers.py (hunk 2 above)
Comment (adrianboguszewski) [reply tools/pot/openvino/tools/pot/api/helpers.py:24]: Agree with @helena-intel

File: tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+            {
+                'type': 'resize',
+                'width': 224,
+                'height': 224
+            }
+        ],
+    }
+
+    # Step 1: Create a user-defined DataLoader object
+    data_loader = ImageNetDataLoader(dataset_config)
```
Comment (adrianboguszewski) [inline tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py:47]: It would be nice to have DataLoaderParameters similarly to QuantizationParameters

**Review by adrianboguszewski — COMMENTED** [review]
This is a great change. It simplifies POT API a lot! 
See also ticket 84295.

File: tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py (hunk 8 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py:47]: This is temporary stuff. We are planning to use framework implementations. No configs in the future, it is supposed to be a Python code.

File: tools/pot/openvino/tools/pot/api/samples/object_detection/object_detection_simple_api.py (hunk 7 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/samples/object_detection/object_detection_simple_api.py:12]: I am not sure what you mean here. If we merge everything into one file in the samples it can be simplified here but will look cumbersome on the sample level.

File: tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py (hunk 4 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py:1]: Sure but I don't think we should do everything in one go. We will discuss this.

File: tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py (hunk 5 above)
Comment (AlexKoff88) [reply tools/pot/openvino/tools/pot/api/samples/classification/classification_simple_api_sample.py:2]: There is a readme for each sample. We will update them.
